io/src/lib.rs

```diff
@@ -7,54 +7,170 @@ use scale_info::TypeInfo;
+    /// *`tokens_metadata` size MUST equal to the length of ids.
+    /// * a sender MUST be an owner or an approved account.
+    ///
+    /// On success returns `MTKEvent::Transfer`
     MintBatch {
+        /// Tokens' IDs to mint.
         ids: Vec<TokenId>,
+        /// Tokens' amounts.
         amounts: Vec<u128>,
+        /// Tokens' metadata.
         tokens_metadata: Vec<Option<TokenMetadata>>,
     },
+
+    /// Transfers token from `from` to `to`.
+    ///
+    /// Requirements:
+    /// * `from` and `to` MUST be different accounts.
+    /// * `from` MUST have sufficient amount of tokens.
+    /// * `to` MUST be a non-zero account.
+    ///
+    /// On success returns `MTKEvent::Transfer`.
     TransferFrom {
+        /// From which account to transfer.
         from: ActorId,
+        /// To which account to transfer.
         to: ActorId,
+        /// Token's ID.
         id: TokenId,
+        /// Token's amount.
         amount: u128,
     },
+
+    /// Transfers multiple tokens from `from` to `to`.
+    ///
+    /// Requirements:
+    /// * `from` and `to` MUST be different accounts.
+    /// * `from` MUST have sufficient amount of tokens.
+    /// * `to` MUST be a non-zero account.
+    /// * `ids` and `amounts` MUST be the same length.
+    ///
+    /// On success returns `MTKEvent::Transfer`.
     BatchTransferFrom {
+        /// From which account to transfer.
         from: ActorId,
+        /// To which account to transfer.
         to: ActorId,
+        /// Tokens' IDs.
         ids: Vec<TokenId>,
+        /// Tokens' amounts.
         amounts: Vec<u128>,
     },
+
+    /// Burns multiple tokens.
+    ///
+    /// # Requirements:
+    /// * a sender MUST have sufficient amount of tokens to burn,
+    /// * a sender MUST be the owner.
+    ///
+    /// On success returns `MTKEvent::Transfer
     BurnBatch {
+        /// Tokens' IDs to burn.
         ids: Vec<TokenId>,
+        /// Tokens' amounts to burn.
         amounts: Vec<u128>,
     },
+
+    /// Allows a `account` to use tokens.
+    ///
+    /// On success returns `MTKEvent::Approval
     Approve {
+        /// Approved account.
         account: ActorId,
     },
+
+    /// Disallows a `account` to use tokens.
+    ///
+    /// On success returns `MTKEvent::RevokeApproval
     RevokeApproval {
+        /// Disapproved account.
         account: ActorId,
     },
+
+    /// Transforms user's tokens to multiple NFTs.
+    ///
+    /// # Requirements:
+    /// * a sender MUST have sufficient amount of tokens to burn,
+    /// * a sender MUST be the owner.
+    ///
+    /// On success returns `MTKEvent::Transfer`.
+    Transform {
+        /// Token's ID to burn.
+        id: TokenId,
+        /// Amount of burnt token.
+        amount: u128,
+        /// To which accounts to mint NFTs.
+        accounts: Vec<ActorId>,
+        /// NFTs' IDs.
+        nft_ids: Vec<Vec<TokenId>>,
+        /// NFTs metadata.
+        nfts_metadata: Vec<Vec<Option<TokenMetadata>>>,
```

Comment:
Why to not combine these three vecs in one vec of structs?

Reply:
Yeah, that's a good call!